### PR TITLE
Add RickRoleMock

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Add a RickRollMock

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

**Review by Insprill — APPROVED** [review]
LGTM

**Comment by seeseemelk** [comment]
I'm personally of the opinion that changes to MockBukkit should aim solely to improve either the experiences of the users that want to test their plugins and/or the maintenance burden for the maintainers.

This PR contributes very little to either. In fact, it provides a net negative - one that it increases our maintenance burden while projecting an image to our downstream users that MockBukkit is more about memes and fun rather than a serious tool used for testing plugins.

I saw this PR a few days ago, and I should have pressed `request changes` instead of simply not approving it until I could figure out if this is something I think would add value.

I've opened a reversal PR which removes these classes as these classes provide nothing extra to our users whilst increasing the size of our codebase needlessly.

**Pull request merged**
